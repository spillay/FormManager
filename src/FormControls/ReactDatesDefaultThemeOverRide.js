import React from 'react'
import ThemedStyleSheet from 'react-with-styles/lib/ThemedStyleSheet';
import aphroditeInterface from 'react-with-styles-interface-aphrodite';
import { css, withStyles } from 'react-with-styles';
import DefaultTheme from 'react-dates/lib/theme/DefaultTheme';

ThemedStyleSheet.registerInterface(aphroditeInterface);
const ReactDatesDefaultThemeOverRide = ThemedStyleSheet.registerTheme({
  reactDates: {
    ...DefaultTheme.reactDates,
    // color: {
    //   ...DefaultTheme.reactDates.color,
    //   // backgroundFocused: '#0187b4',
    //   placeholderText: '#0187b4',
    // },
    sizing: {
      inputWidth: 105,
      arrowWidth: 20,
    },

    font: {
      size: 12,
      captionSize: 12,
      input: {
        size: 13,
        lineHeight: '13px',
        size_small: 13,
        lineHeight_small: '13px',

      },
    },

  },
});

export { css, withStyles, ReactDatesDefaultThemeOverRide };