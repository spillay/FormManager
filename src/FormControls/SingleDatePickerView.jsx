import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { SingleDatePicker } from 'react-dates';
import moment from 'moment';

export class SingleDatePickerView extends React.Component {

    constructor(props) {
        super(props);
        //this.state = {date:moment()}
        this.state = {date:null}
    }
    onDateChange = (date) => {
        // console.log(`onDateChange key :${this.props._key}`," date ", date)
        this.setState({date:date});
        this.props.onChange({target:{value:date.format("MMM Do YYYY")}},this.props._key)
    } 
    onFocusChange = ({ focused }) => {
        this.setState({ focused })
    }

    render() {
        return (
            <div key={this.props._key} >
                <SingleDatePicker
                    date={this.state.date} // momentPropTypes.momentObj or null  current date
                    onDateChange={this.onDateChange} // PropTypes.func.isRequired
                    focused={this.state.focused} // PropTypes.bool
                    onFocusChange={this.onFocusChange}
                    {...this.props.options}                 
                />
                <div className="validation-error" >{this.props.errorFor(this.props._key)}</div>
                <small className="form-text text-muted ">( Please provide valid {this.props.label}. )</small>
            </div>
        )
    }


} 
SingleDatePickerView.propTypes = {
    _key: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    errorFor: PropTypes.func.isRequired
}
