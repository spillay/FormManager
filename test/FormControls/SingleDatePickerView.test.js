import React from 'react';
import { SingleDatePickerView } from '../../src/FormControls/SingleDatePickerView'
import renderer from 'react-test-renderer';

test('SingleDatePickerView test', () => {
    const baseProps = {
        _key: "Single",
        label: "Date Single",
        onChange: () => { console.log("onChange") },
        errorFor: () => { console.log("Error") },
    };
    const component = renderer.create(
        <SingleDatePickerView {...baseProps} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    //   // manually trigger the callback
    //   tree.props.onMouseEnter();
    //   // re-rendering
    //   tree = component.toJSON();
    //   expect(tree).toMatchSnapshot();

    //   // manually trigger the callback
    //   tree.props.onMouseLeave();
    //   // re-rendering
    //   tree = component.toJSON();
    //   expect(tree).toMatchSnapshot();
});