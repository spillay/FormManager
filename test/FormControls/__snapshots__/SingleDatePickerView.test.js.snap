// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleDatePickerView test 1`] = `
<div>
  <div
    className="SingleDatePicker SingleDatePicker_1"
  >
    <div>
      <div
        className="SingleDatePickerInput SingleDatePickerInput_1 SingleDatePickerInput__withBorder SingleDatePickerInput__withBorder_2"
      >
        <div
          className="DateInput DateInput_1"
        >
          <input
            aria-describedby="DateInput__screen-reader-message-date"
            aria-label="Date"
            autoComplete="off"
            className="DateInput_input DateInput_input_1"
            disabled={false}
            id="date"
            name="date"
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Date"
            readOnly={false}
            required={false}
            type="text"
            value=""
          />
          <p
            className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
            id="DateInput__screen-reader-message-date"
          >
            Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="validation-error"
  />
  <small
    className="form-text text-muted "
  >
    ( Please provide valid 
    Date Single
    . )
  </small>
</div>
`;
